library/common-library/templates/_images.tpl

```diff
@@ -0,0 +1,53 @@
+{{/*
+Return the proper image name
+{{ include "common.images.image" ( dict "imageRoot" .Values.path.to.the.image "context" .) }}
+*/}}
+{{- define "common.images.image" -}}
+{{- $registryName := "" -}}
+{{- $repositoryName := .imageRoot.repository -}}
+{{- $tag := .imageRoot.tag | default .context.Chart.AppVersion | toString -}}
+
+{{- if .context.Values.global }}
+    {{- if .context.Values.global.image.registry }}
+     {{- $registryName = .context.Values.global.image.registry  -}}
+    {{- end -}}
+{{- end -}}
+
+{{- with .imageRoot.registry }}
+    {{- $registryName = . -}}
+{{ end }}
```

Comment:
This logic that allows superseding the global value with a local one is already done in the `lowDataMode` helper: https://github.com/newrelic/helm-charts/blob/feat/imageGlobals/library/common-library/templates/_low-data-mode.tpl

Using that logic allows us in the future to simply put `{{ include "common.images.image" . }}` instead of having to create a `dict` for each helper.

It could sound that this is not a big deal now because we have 3 or 4 helpers but when we have 20 we are going to start finding issues like which dictionary does this helper needs to render correctly.

Is a pattern that was useful in the compatibility layer but I do not see that is going to work here.

Reply:
The issue is that with the images helpers is that we cannot assume where fetching the image, therefore calling the function we need to pass the root of the image

Reply:
I think this approach is overcomplicating because of the sidecars.

I would create a helper for the registry, for the repository and the tag and a fourth that concatenates them.

In the Helm charts where we use sidecars we could opt to not use the common function that concatenated them and use the `.Values.images.agent.repository` if we need to.